kratos/containers/master_slave_relation.h

```diff
@@ -0,0 +1,201 @@
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//
+//
+
+#if !defined(CONSTRAINT_EQUATION_H)
+#define CONSTRAINT_EQUATION_H
+// System includes
+#include <vector>
+#include <unordered_set>
+#include <iostream>
+#include <assert.h>
+
+// project includes
+#include "includes/define.h"
+#include "includes/dof.h"
+#include "includes/node.h"
+#include "includes/process_info.h"
+
+namespace Kratos
+{
+
+
+/**
+ * @class MasterSlaveRelation
+ * @ingroup KratosCore
+ * @brief This class stores the information regarding the MasterSlaveRelation equation.
+ *         Naming convenction is defined like this. (each object of this class will store one equation in the given form
+ *
+ *   SlaveEquationId = w_1*MasterEquationId_1 + w_2*MasterEquationId_2 + ..... + w_n*MasterEquationId_n
+ *
+ *   This stores the condensed form of the MasterSlaveConstraint objects into one object. if only one relation for a slave is added as
+ *   MasterSlaveConstraint then there will only be one entry for master for its corresponding MasterSlaveRelation.
+ *   Currently this class is designed to hold only one equation.
+ *
+ *   Future plan is to also make it possible to work with matrices (T) and vectors (for slave and master equation ids and constants)
+ *
+ *
+ *  IMPORTANT : This is not seen by the user. This is a helper data structure which is created in the builder and solver and destroyed there.
+ *
+ * @author Aditya Ghantasala
+ */
+class MasterSlaveRelation : public IndexedObject
+{
+  public:
+    typedef std::size_t IndexType;
+    typedef Matrix MatrixType;
+    typedef Vector VectorType;
+    typedef std::vector<IndexType> EquationIdVectorType;
+    KRATOS_CLASS_POINTER_DEFINITION(MasterSlaveRelation);
+
+    // empty constructor and methods to add master and slave independently.
+    MasterSlaveRelation() : IndexedObject(0)
+    {
+        SetConstant(0.0);
+        //SetConstantUpdate(0.0);
+    }
+
+
+    MasterSlaveRelation(IndexType const &rSlaveEquationId) : IndexedObject(rSlaveEquationId)
+    {
+        SetConstant(0.0);
+        //SetConstantUpdate(0.0);
+    }
+
+    void SetConstant(double Constant) { mConstant = Constant; }
+    //void SetConstantUpdate(double ConstantUpdate) { mConstantUpdate = ConstantUpdate; }
+    double Constant() const { return mConstant; }
+    //double ConstantUpdate() const { return mConstantUpdate; }
+    IndexType SlaveEquationId() const { return this->Id(); }
+
+
+
+    // Get number of masters for this slave
+    std::size_t GetNumberOfMasters() const
+    {
+        return mMasterDataSet.size();
+    }
+
+
+    /**
+     * this determines the master equation IDs connected to this constraint
+     * @param rResult the elemental equation ID vector
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    virtual void EquationIdVector(EquationIdVectorType& rSlaveEquationIds,
```

Comment:
if you know that you can have just one slave, just return one id and not a vector

Reply:
In that case, the builder and solver which is using this function will have to implement two different versions : one where only one equation Id of slave is returned, or a vector is returned ? 
Wouldn't it become a bit cumbersome to implement this ?